### PR TITLE
Add Integrated Gradients interpretation method and minor refinements

### PR DESCRIPTION
**Pull request by Mdnaimulislam**
### Description
Added Integrated Gradients method for interpretation and updated checkpoint remapping for model parameter names. Additionally, renamed ImageVaeDecoder to ImageVAEDecoder for consistency.

### Status
**Ready**

### Types of changes
- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [x] In-line docstrings updated.
- [ ] [Source for documentation at `docs`](https://github.com/pykale/pykale/tree/main/docs/source) manually updated for new API.

### CODE REVIEW

File: kale/interpret/signal_image_attribution.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+    attributions_image = attributions[0]
+    attributions_signal = attributions[1]
+
+    # --- Signal Attribution ---
+    attributions_signal_np = attributions_signal.cpu().detach().numpy().squeeze()
+    norm_attributions_signal = (attributions_signal_np - attributions_signal_np.min()) / (
+        attributions_signal_np.max() - attributions_signal_np.min() + 1e-8
+    )
+    signal_waveform_np = signal_smoothed_tensor.cpu().detach().numpy().squeeze()
+    full_length = min(60000, len(signal_waveform_np))
```
Comment (shuo-zhou) [inline kale/interpret/signal_image_attribution.py]: Why choose 60000 (and 6000 below)? At lease add in-line comments

File: kale/utils/remap_model_parameters.py
Comment (shuo-zhou) [inline kale/utils/remap_model_parameters.py:1]: Is it better to move this file to `dataprep` and rename this API as `dict_prep`?

**Review by shuo-zhou — APPROVED** [review]
See my comments and let me know if you need a discussion

File: kale/utils/remap_model_parameters.py
Comment (Mdnaimulislam) [reply kale/utils/remap_model_parameters.py:1]: No, this is not connected with data at all. It's connected with the model's parameter name. This is a helper function that can be used with any other models.

File: tests/utils/test_remap_model_parameters.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
+# =============================================================================
+# Author: Mohammod Suvon, m.suvon@sheffield.ac.uk
+# =============================================================================
+
+import pytest
+
+from kale.utils.remap_model_parameters import remap_state_dict_keys  # Change this to your actual module path
```
[the remark this answers is not in the document]
Comment (Mdnaimulislam) [reply tests/utils/test_remap_model_parameters.py]: fixed.

File: kale/utils/remap_model_parameters.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+    Returns:
+        dict:
+            A new state dictionary with parameter names remapped according to the defined mapping.
+    """
+    mapping = [
+        ("ecg_encoder.", "signal_encoder."),
+        ("ecg_decoder.", "signal_decoder."),
+        ("fc_logvar", "fc_log_var"),
+        ("signal_encoder.fc_logvar.weight", "signal_encoder.fc_log_var.weight"),
```
[the remark this answers is not in the document]
Comment (Mdnaimulislam) [reply kale/utils/remap_model_parameters.py]: fixed.

File: tests/predict/test_decode.py [hunk 4]
```diff
@@ -128,9 +128,9 @@ def test_image_vae_decoder_forward():
 
 
 def test_image_vae_decoder_repr_and_init():
-    decoder = ImageVaeDecoder(latent_dim=16, output_channels=3)
-    assert isinstance(decoder, ImageVaeDecoder)
-    assert "ImageVaeDecoder" in repr(decoder) or "ImageVAEDecoder" in repr(decoder)
+    decoder = ImageVAEDecoder(latent_dim=16, output_channels=3)
+    assert isinstance(decoder, ImageVAEDecoder)
+    assert "ImageVAEDecoder" in repr(decoder) or "ImageVAEDecoder" in repr(decoder)
```
[the remark this answers is not in the document]
Comment (Mdnaimulislam) [reply tests/predict/test_decode.py]: fixed.

File: kale/interpret/signal_image_attribution.py [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+            - x_pts, y_pts : np.ndarray
+                Coordinates of important points in the image (after dilation).
+            - importance_pts : np.ndarray
+                Attribution values at (x_pts, y_pts).
+            - signal_threshold : float
+                The threshold used for signal attributions.
+            - image_threshold : float
+                The threshold used for image attributions.
+    """
+    batches = list(last_val_loader)
```
[the remark this answers is not in the document]
Comment (Mdnaimulislam) [reply kale/interpret/signal_image_attribution.py:91]: No need for interpretation, at most we pass a single batch of the dataset and not the entire data loader.

File: setup.py [hunk 6]
```diff
@@ -60,6 +60,9 @@
     "xlsxwriter",
     "prettytable",
     "comet_ml",
+    "neurokit2",
+    "captum",
+    "PyWavelets",
```
[the remark this answers is not in the document]
Comment (Mdnaimulislam) [reply setup.py]: fixed

File: kale/interpret/signal_image_attribution.py (hunk 1 above)
Comment (Mdnaimulislam) [reply kale/interpret/signal_image_attribution.py]: Changed 60000 to int(lead_number * sampling_rate * signal_length) to represent the total signal length.

Changed 6000 to int(lead_number * sampling_rate) to obtain the zoomed-in values in seconds rather than in data points.

File: kale/utils/remap_model_parameters.py
Comment (Mdnaimulislam) [reply kale/utils/remap_model_parameters.py:1]: In future, maybe make it more genral function to be included in dataprep.
